src/main/java/org/jpeek/Report.java

```diff
@@ -130,15 +139,37 @@
      * @param mean Mean
      * @param sigma Sigma
      * @checkstyle ParameterNumberCheck (10 lines)
+     * @todo #135:30min Make Report save 'metric.xsd' itself. Currently, Report
+     *  is setting the schema's location on the output metric XML based on the
+     *  assumption that App will always save it to a relative subdirectory
+     *  'xsd' and with the same filename 'metric.xsd'. This promise is best
+     *  kept if Report class has the responsibility of saving metric.xsd itself.
      */
+    @SuppressWarnings("unchecked")
     Report(final XML xml, final String name,
         final Map<String, Object> args,
         final double mean, final double sigma) {
         this.skeleton = xml;
         this.metric = name;
-        this.params = args;
+        this.params = new MapOf<>(
+            args,
+            new MapEntry<>("schemaLocation", Report.SCHEMA_FILE)
+        );
         this.post = new XSLChain(
             new CollectionOf<>(
+                new XSLDocument(
+                    new UncheckedText(
+                        new TextOf(
+                            new InputOf(
+                                Report.class.getResourceAsStream(
+                                    "xsl/metric-post-schemaloc.xsl"
```

Comment:
@llorllale looks weird. Can't we just add this single attribute to the `skeleton.xml` when we are building it? Why making this simple operation so complex?

Reply:
@yegor256 My line of reasoning is: `Report` creates `metric.xml` so it is responsible for properly building it.

`Skeleton` creates the `skeleton.xml` so I believe it is responsible for properly building that xml.

I see that `App` is currently creating both `index.xml` and `matrix.xml` and I think those two things should be extracted into their own classes who in turn will be responsible for properly building those xmls.

WDYT?

Reply:
@yegor256 let's also remember that the value for `xsi:noNamespaceSchemaLocation` will be different among the files.

Reply:
@llorllale why can't you set it via Xembly? Why using XSL?

Reply:
@yegor256 

> why can't you set it via Xembly?

`Xembler` works with nodes. Can't set the required attribute on the node returned by the `XML` resulting from transforming the skeleton into `metric` because it's a document node (throws error "Cannot cast ...DocumentImpl to ...Element"). I think I would have to bend over backwards in the code to make it work.

> Why using XSL?

We already have a post-processing stage in `Report` so I took advantage of that. Also, this way seems cleaner to me than the alternative I described above. The noise can be reduced a little if `XSLDocument` had a ctor that accepted `InputStream` and `params` together.